code/autoannotate.py

```diff
@@ -0,0 +1,321 @@
+                                  -o <destination_dir>
+                                  -r <reference taxonomy>
+                                  -p <string separator>
+                                  -g <genus placeholder>
+                                  -s <species placeholder>
+
+    Generate full taxonomy strings from a reference database, given
+    a list of "source" genus and species names.
+
+    -i / --infile: filepath
+        tab-separated list of genus/species names and [optionally]
+        relative abundances in format:
+        Taxonomy    Sample1
+        Lactobacillus plantarum 0.5
+        Pediococcus damnosus    0.5
+
+    -o / --outfile: path
+        directory in which to write annotated taxonomy file
+
+    -r / --ref_taxa: filepath
+        tab-separated list of semicolon-delimited taxonomy strings
+        associated with reference sequences. In format:
+        seqID   taxonomy
+        0001    kingdom;phylum;class;order;family;genus;species
+
+    -p / --separator: str
+        taxonomy strings are separated with this string pattern.
+        Default = ";"
+
+    -g / --genus: str
+        Placeholder to use for taxa that have no genus-level match
+        in reference taxonomy file. Should match the conventions
+        that are used in that reference taxonomy file.
+        Default = "g__"
+
+    -s / --species: str
+        Placeholder to use for taxa that have no species-level match
+        in reference taxonomy file. Should match the conventions
+        that are used in that reference taxonomy file.
+        Default = "s__"
+
+    '''
+
+
+def main(argv):
+    source_fp = ''
+    destination_dir = ''
+    ref_taxa_fp = ''
+    sep = ';'
+    gen = 'g__'
+    sp = 's__'
+
+    try:
+        opts, args = getopt.getopt(argv, "hi:o:r:p:g:s:", ["help",
+                                                           "infile=",
+                                                           "outdir=",
+                                                           "ref_taxa=",
+                                                           "separator",
+                                                           "genus",
+                                                           "species"])
+    except getopt.GetoptError:
+        print(usage)
+        sys.exit(2)
+
+    for opt, arg in opts:
+        if opt in ("-h", "--help"):
+            print(usage)
+            sys.exit()
+        elif opt in ("-i", "--infile"):
+            source_fp = arg
+        elif opt in ("-o", "--outdir"):
+            destination_dir = arg
+        elif opt in ("-r", "--ref_taxa"):
+            ref_taxa_fp = arg
+        elif opt in ("-p", "--separator"):
+            sep = arg
+        elif opt in ("-g", "--genus"):
+            gen = arg
+        elif opt in ("-s", "--species"):
+            sp = arg
+
+    # generate dict of {name: (genus, species, abundances)}
+    with open(source_fp, "r") as source:
```

Comment:
If `source_fp` doesn't exist, the user is going to see a traceback error. You should detect these types of issues and present more useful error messages.